File: frontend/src/apis/merchant/GetMerchantStoreImageApi.ts
import type { FileNameText } from '@/objects/core/SharedObjects'
import { API_CLIENT_DEFAULTS } from '@/system/api/ApiConstants'
import { buildApiPath1, defineJsonGetApi1, routeSegment } from '@/system/api/TypedApiDefinitions'

const getMerchantStoreImageApiDefinition = defineJsonGetApi1<FileNameText, void>(
  [routeSegment('uploads'), routeSegment('store-images')],
)

export function getMerchantStoreImageUrl(filename: FileNameText) {
  return `${API_CLIENT_DEFAULTS.backendUrl}${buildApiPath1(getMerchantStoreImageApiDefinition, filename)}`
}

File: backend/src/main/scala/services/merchant/api/GetMerchantStoreImageApi.scala
package services.merchant.api

import domain.shared.given

import domain.shared.FileNameText
import system.api.*

val getMerchantStoreImageApi: FixedMethodApi1[FileNameText, Unit] =
  jsonGetApi1[FileNameText, Unit](
    List(routeSegment("uploads"), routeSegment("store-images")),
  )
